README.md

```diff
@@ -355,6 +356,62 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+#### Preferring Interfaces Over Intersections
```

Comment:
After talking with @mmnathan, we would prefer using types over interfaces, unless:
* You need a feature that is only supported by interface (e.g. declaration merging)
* your project becomes so huge and slow that using interfaces could speed up things